### PR TITLE
Small fixes to Lit-GPT demo

### PR DESCRIPTION
**Pull request by gkroiz**
- Define Data and Experiment GCS buckets separately 
- clone lit-gpt at specific commit (https://github.com/Lightning-AI/lit-gpt/commit/d5d371417ecb3d3b6c4f30837d8bb7cf2b5310ae)
- fix location of `out/` folder so that it syncs with gcs buckets

### CODE REVIEW

File: sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {{- $requiredVar := .Values.cluster.nNodes | required ".Values.cluster.nNodes is required" -}}
 {{- $requiredVar := .Values.cluster.nodePool | required ".Values.cluster.nodePool is required" -}}
 {{- $requiredVar := .Values.network.ncclIfnames | required ".Values.ncclIfnames is required" -}}
-{{- $requiredVar := .Values.workload.gcsBucket | required ".Values.gcsBucket is required" -}}
 {{- $requiredVar := .Values.workload.jobTimestamp | required ".Values.jobTimestamp is required" -}}
+{{- $requiredVar := .Values.workload.gcsExperimentBucket | required ".Values.gcsExperimentBucket is required" -}}
```
Comment (Chris113113) [inline sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml:5]: Thoughts on calling this LogBucket instead of ExperimentBucket? AFAIK we're only dumping logs in here.

File: sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml [hunk 2]
```diff
@@ -145,18 +146,20 @@ spec:
...
+        value: "{{$root.Values.workload.gcsDataBucket}}"
       - name: DATA_DIR
         value: "{{$root.Values.workload.dataDir}}"
       - name: BATCH_SIZE
         value: "{{$root.Values.workload.batchSize}}"
       - name: MICRO_BATCH_SIZE
         value: "{{$root.Values.workload.microBatchSize}}"
       - name: MODEL_NAME
-        value: "{{$root.Values.workload.modelName}}"
+        value: "pythia-6.9b"
```
Comment (Chris113113) [inline sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml:162]: Undo this change, hardcoding model name.

**Review by Chris113113 — CHANGES_REQUESTED** [review]

File: sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml (hunk 2 above)
Comment (gkroiz) [reply sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml]: Fixed

File: sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml (hunk 1 above)
Comment (gkroiz) [reply sample_workloads/lit-gpt-demo/helm/templates/litgpt.yaml:5]: This is true, with the exception of `metrics.csv` with includes performance metrics. I think ExperimentBucket is more fitting because of this.

**Review by Chris113113 — APPROVED** [review]
